app/src/main/java/com/medmapper/v33001/MainViewModel.kt

```diff
@@ -67,14 +67,14 @@ class MainViewModel(var medicineService : IMedicineService) /*= MedicineService(
         user?.let {
             user ->
             val document =
-                if (selectedMedicine.medicationID == null || selectedMedicine.medicationID.isEmpty()) {
+                if (selectedMedicine.id == null || selectedMedicine.id.isEmpty()) {
                     // create a new medicine
                     firestore.collection("users").document(user.uid).collection("medications").document()
                 } else {
                     // update an existing specimen
                     firestore.collection("users").document(user.uid).collection("medications").document()
                 }
-            selectedMedicine.medicationID = document.id
+            selectedMedicine.id = document.id
```

Comment:
Changed selectedMedicine.medicationID to id to reflect the existing constructor/attribute that is in the Medicine DTO

Reply:
Either one works.